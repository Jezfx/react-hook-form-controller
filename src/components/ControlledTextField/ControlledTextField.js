import React from "react";
import { Controller } from "react-hook-form";
import { TextField, FormControl, FormLabel } from "@material-ui/core";

import { useFormContext } from "react-hook-form";

const ControlledTextField = ({
  name,
  label,
  defaultValue,
  helperText,
  ...rest
}) => {
  const { control, errors } = useFormContext();
  const error = errors[name];

  return (
    <FormControl hiddenLabel={true} margin="none" error={!!error}>
      {label && <FormLabel htmlFor={name}>{label}</FormLabel>}
      <Controller
        name={name}
        control={control}
        defaultValue={defaultValue}
        render={({ onChange, value = "", name }) => (
          <TextField
            error={!!error}
            name={name}
            value={value}
            onChange={onChange}
            variant="outlined"
            helperText={error?.message || helperText}
            {...rest}
          />
        )}
      />
    </FormControl>
  );
};

export default ControlledTextField;
