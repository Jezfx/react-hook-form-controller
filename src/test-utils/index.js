import { render } from "@testing-library/react";
import { useForm, FormProvider } from "react-hook-form";

export const renderWithProviders = (ui, { defaultValues = {} } = {}) => {
  const Wrapper = ({ children }) => {
    const methods = useForm({});
    return <FormProvider {...methods}>{children}</FormProvider>;
  };
  return render(ui, { wrapper: Wrapper });
};
