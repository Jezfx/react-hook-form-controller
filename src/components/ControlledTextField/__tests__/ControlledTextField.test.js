import React from "react";
import ControlledTextField from "../";
import { renderWithProviders } from "../../../test-utils";

describe("ControlledTextField", () => {
  it("should render correctly", () => {
    const { asFragment } = renderWithProviders(
      <ControlledTextField name="foo" label="Foo" />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
