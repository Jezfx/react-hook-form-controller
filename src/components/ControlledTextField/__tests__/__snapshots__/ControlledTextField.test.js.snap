// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlledTextField should render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
      data-shrink="false"
    >
      Foo
    </label>
    <div
      class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiFilledInput-input"
        name="foo"
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;
